### PR TITLE
add docs about writing / generating docs

### PR DESCRIPTION
**Pull request by jakebolewski**
* fix some broken external links
* don't deploy by default in local environments when building docs

I have

- [ ] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [ ] Followed all necessary [style guidelines](https://CliMA.github.io/CLIMA/latest/CodingConventions.html) and run `julia .dev/climaformat.jl .`
- [ ] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [ ] There are no open pull requests for this already
- [ ] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://CliMA.github.io/CLIMA/latest/CodingConventions.html) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

File: docs/make.jl [hunk 1]
```diff
@@ -56,8 +56,11 @@ makedocs(
 
 include("clean_build_folder.jl")
 
-deploydocs(
-    repo = "github.com/CliMA/ClimateMachine.jl.git",
-    target = "build",
-    push_preview = true,
-)
+
+if get(ENV, "CLIMATEMACHINE_CI_DEPLOY_DOCS", "") == "true"
```
Comment (charleskawczynski) [inline docs/make.jl:60]: Is this necessary? I figured that there is some internal check. Also, there's a [deploy_config](https://juliadocs.github.io/Documenter.jl/stable/lib/public/#Documenter.deploydocs) kwarg. I think our GHA docs check for `ENV["DOCUMENTER_KEY"]`.

File: docs/make.jl [hunk 2]
```diff
@@ -45,7 +45,7 @@ makedocs(
     sitename = "ClimateMachine",
     doctest = false,
     strict = false,
-    linkcheck = false,
+    linkcheck = true,
```
Comment (charleskawczynski) [inline docs/make.jl:48]: ❤️

File: docs/src/Contributing.md [hunk 3]
```diff
@@ -96,3 +96,27 @@ We use [`bors`](https://bors.tech/) to manage the `ClimateMachine` repo.
 If you're a collaborator and have the necessary permissions, you can type
 `bors try` in a comment on a PR to have some additional tests run on that
 PR, or `bors r+` to try and merge the code.
+
+## Contributing Documentation
+
+Documentation is written Julia flavored markdown and generated from two sources:
```
Comment (charleskawczynski) [inline docs/src/Contributing.md:102]:
```suggestion
Documentation is written in Julia-flavored markdown and generated from two sources:
```

File: docs/make.jl (hunk 1 above)
Comment (jakebolewski) [reply docs/make.jl:60]: hmm no, but I do get an exception with the current default behaviour?  it looks like it's picking up `GITHUB_TOKEN` env variable for deployment.

Comment (charleskawczynski) [reply docs/make.jl:60]: Strange, what’s the exception?

Comment (simonbyrne) [reply docs/make.jl:60]: I agree, this shouldn't be necessary. It also breaks the docs preview.

**Pull request merged**